subprojects/core-api/src/main/java/org/gradle/api/artifacts/transform/TransformOutputs.java

```diff
@@ -35,7 +35,31 @@
 public interface TransformOutputs {
     /**
      * Registers an output directory.
-     * For a relative path, a location for the output directory is provided.
+     *
+     * <p>
+     *     For an <strong>absolute path</strong>, the location is registered as an output directory of the {@link TransformAction}.
+     *     The path is required to point to the {@link InputArtifact}, or point inside the input artifact if it is a directory.
+     *     Example:
+     * </p>
+     * <pre>
```

Comment:
Can we make this auto-tested?

Reply:
I added auto-testing for this class. It adds quite some boilerplate. Do you think it is worth the effort?